### PR TITLE
Genjobs: support reading from a yaml configuration file(s)

### LINKED ISSUE #2161: Genjobs: additionally support a yaml configuration file
As the number of jobs generated by [`genjobs`](https://github.com/istio/test-infra/tree/master/prow/genjobs) grow, so does the complexity in the command-line invocations and their respective arguments (e.g. [`gen-private-jobs.sh`](https://github.com/istio/test-infra/blob/master/prow/genjobs/gen-private-jobs.sh)). 

While having a CLI interface is *great* for simple use-cases, additionally providing a configuration file as input (similar to [`generate.go`](https://github.com/istio/test-infra/blob/master/prow/config/generate.go)) can greatly reduce complexity and maximize utility of the tool.

### PR DESCRIPTION
**Pull request by clarketm**
fix https://github.com/istio/test-infra/issues/2161 and also enabled RBE on rest of `istio-private/proxy` jobs.

> In the near future, `genjobs` will be subsumed into the `transform` subcommand of the [ProwJob CLI](https://docs.google.com/document/d/1bUNo7mpERzwMerDkqwB2dA2PMThwURgLyoIr1Awse5w/edit?usp=sharing) (draft).

### CODE REVIEW

File: prow/config/istio-private_jobs/00-clean.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+common_opts: &common_opts
```
Comment (howardjohn) [inline prow/config/istio-private_jobs/00-clean.yaml:1]: why do we need this if its only used one time?

File: prow/config/istio-private_jobs/istio.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+common_jobs_mods: &common_jobs_mods
```
Comment (howardjohn) [inline prow/config/istio-private_jobs/istio.yaml:1]: if we want to avoid yaml magic, you could add a native `defaults` field that does this. Up to you

**Review by howardjohn — APPROVED** [review]

File: prow/config/istio-private_jobs/00-clean.yaml (hunk 1 above)
Comment (clarketm) [reply prow/config/istio-private_jobs/00-clean.yaml:1]: You are right, it is not needed. I originally had all the transforms in a single file. I missed that this particular anchor was used only once.

File: prow/config/istio-private_jobs/istio.yaml (hunk 2 above)
Comment (clarketm) [reply prow/config/istio-private_jobs/istio.yaml:1]: I like the idea. I am currently doing research into various differents yaml processors, mergers, and templating tools (e.g. kustomize, helm, quicktemplate, ytt, aviator, jsonnet, yq, spruce, etc.) that may help simply the configuration.

**Pull request merged**